File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-valuation-5-4/Valuations.cpp
// Valuations.cpp 
#include "Valuations.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Valuations::Valuations(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //valuationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationNode = xmlNode->FirstChildElement("valuation");

   if(valuationNode){valuationIsNull_ = false;}
   else{valuationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationNode , address : " << valuationNode << std::endl;
   #endif
   if(valuationNode)
   {
      if(valuationNode->Attribute("href") || valuationNode->Attribute("id"))
      {
          if(valuationNode->Attribute("id"))
          {
             valuationIDRef_ = valuationNode->Attribute("id");
             valuation_ = boost::shared_ptr<AssetValuation>(new AssetValuation(valuationNode));
             valuation_->setID(valuationIDRef_);
             IDManager::instance().setID(valuationIDRef_,valuation_);
          }
          else if(valuationNode->Attribute("href")) { valuationIDRef_ = valuationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuation_ = boost::shared_ptr<AssetValuation>(new AssetValuation(valuationNode));}
   }

   //valuationReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationReferenceNode = xmlNode->FirstChildElement("valuationReference");

   if(valuationReferenceNode){valuationReferenceIsNull_ = false;}
   else{valuationReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationReferenceNode , address : " << valuationReferenceNode << std::endl;
   #endif
   if(valuationReferenceNode)
   {
      if(valuationReferenceNode->Attribute("href") || valuationReferenceNode->Attribute("id"))
      {
          if(valuationReferenceNode->Attribute("id"))
          {
             valuationReferenceIDRef_ = valuationReferenceNode->Attribute("id");
             valuationReference_ = boost::shared_ptr<ValuationReference>(new ValuationReference(valuationReferenceNode));
             valuationReference_->setID(valuationReferenceIDRef_);
             IDManager::instance().setID(valuationReferenceIDRef_,valuationReference_);
          }
          else if(valuationReferenceNode->Attribute("href")) { valuationReferenceIDRef_ = valuationReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuationReference_ = boost::shared_ptr<ValuationReference>(new ValuationReference(valuationReferenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AssetValuation> Valuations::getValuation()
{
   if(!this->valuationIsNull_){
        if(valuationIDRef_ != ""){
             return boost::shared_static_cast<AssetValuation>(IDManager::instance().getID(valuationIDRef_));
        }else{
             return this->valuation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AssetValuation>();
   }
}
boost::shared_ptr<ValuationReference> Valuations::getValuationReference()
{
   if(!this->valuationReferenceIsNull_){
        if(valuationReferenceIDRef_ != ""){
             return boost::shared_static_cast<ValuationReference>(IDManager::instance().getID(valuationReferenceIDRef_));
        }else{
             return this->valuationReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ValuationReference>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-valuation-5-4/Valuations.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Valuations : ISerialized
    {
        public Valuations(XmlNode xmlNode)
        {
            XmlNode valuationNode = xmlNode.SelectSingleNode("valuation");
            
            if (valuationNode != null)
            {
                if (valuationNode.Attributes["href"] != null || valuationNode.Attributes["id"] != null) 
                {
                    if (valuationNode.Attributes["id"] != null) 
                    {
                        valuationIDRef_ = valuationNode.Attributes["id"].Value;
                        AssetValuation ob = new AssetValuation(valuationNode);
                        IDManager.SetID(valuationIDRef_, ob);
                    }
                    else if (valuationNode.Attributes["href"] != null)
                    {
                        valuationIDRef_ = valuationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuation_ = new AssetValuation(valuationNode);
                    }
                }
                else
                {
                    valuation_ = new AssetValuation(valuationNode);
                }
            }
            
        
            XmlNode valuationReferenceNode = xmlNode.SelectSingleNode("valuationReference");
            
            if (valuationReferenceNode != null)
            {
                if (valuationReferenceNode.Attributes["href"] != null || valuationReferenceNode.Attributes["id"] != null) 
                {
                    if (valuationReferenceNode.Attributes["id"] != null) 
                    {
                        valuationReferenceIDRef_ = valuationReferenceNode.Attributes["id"].Value;
                        ValuationReference ob = new ValuationReference(valuationReferenceNode);
                        IDManager.SetID(valuationReferenceIDRef_, ob);
                    }
                    else if (valuationReferenceNode.Attributes["href"] != null)
                    {
                        valuationReferenceIDRef_ = valuationReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuationReference_ = new ValuationReference(valuationReferenceNode);
                    }
                }
                else
                {
                    valuationReference_ = new ValuationReference(valuationReferenceNode);
                }
            }
            
        
        }
        
    
        #region Valuation_
        private AssetValuation valuation_;
        public AssetValuation Valuation_
        {
            get
            {
                if (this.valuation_ != null)
                {
                    return this.valuation_; 
                }
                else if (this.valuationIDRef_ != null)
                {
                    valuation_ = IDManager.getID(valuationIDRef_) as AssetValuation;
                    return this.valuation_; 
                }
                else
                {
                      return this.valuation_; 
                }
            }
            set
            {
                if (this.valuation_ != value)
                {
                    this.valuation_ = value;
                }
            }
        }
        #endregion
        
        public string valuationIDRef_ { get; set; }
        #region ValuationReference_
        private ValuationReference valuationReference_;
        public ValuationReference ValuationReference_
        {
            get
            {
                if (this.valuationReference_ != null)
                {
                    return this.valuationReference_; 
                }
                else if (this.valuationReferenceIDRef_ != null)
                {
                    valuationReference_ = IDManager.getID(valuationReferenceIDRef_) as ValuationReference;
                    return this.valuationReference_; 
                }
                else
                {
                      return this.valuationReference_; 
                }
            }
            set
            {
                if (this.valuationReference_ != value)
                {
                    this.valuationReference_ = value;
                }
            }
        }
        #endregion
        
        public string valuationReferenceIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

